src/actions/Entities/NameActions.js

```diff
@@ -42,13 +43,12 @@ const update = (id, field, value) => ({
 const updatePatient = detailsEntered => (dispatch, getState) => {
   const currentPatient = selectEditingName(getState());
   const { id } = currentPatient || {};
-  const currentKeys = Object.keys(currentPatient);
 
   if (!id) return;
 
   batch(() => {
     Object.entries(detailsEntered).forEach(([key, value]) => {
-      if (currentKeys.includes(key) && currentPatient[key] !== value) {
+      if (currentPatient[key] !== value) {
```

Comment:
wasn't sure what this was really trying to do - but it stopped me from updating when for example I started with a 'default name' record and went to edit values like gender which weren't in the default name yet